src/py4vasp/calculation/_partial_charge.py

```diff
@@ -0,0 +1,434 @@
+    Since this is postprocessing data for a fixed density, there are no ionic steps
+    to separate the data.
+    """
+
+    @property
+    def stm_settings(self):
+        return STM_settings()
+
+    @_base.data_access
+    def __str__(self):
+        """Return a string representation of the partial charge density."""
+        return f"""
+        {"spin polarized" if self._spin_polarized() else ""} partial charge density of {self._topology()}:
+        on fine FFT grid: {self.grid()}
+        {"summed over all contributing bands" if 0 in self.bands() else f" separated for bands: {self.bands()}"}
+        {"summed over all contributing k-points" if 0 in self.kpoints() else f" separated for k-points: {self.kpoints()}"}
+        """.strip()
+
+    @_base.data_access
+    def grid(self):
+        return self._raw_data.grid[:]
+
+    @_base.data_access
+    def to_dict(self):
+        """Store the partial charges in a dictionary.
+
+        Returns
+        -------
+        dict
+            The dictionary contains the partial charges as well as the structural
+            information for reference.
+        """
+
+        parchg = np.squeeze(self._raw_data.partial_charge[:].T)
+        return {
+            "structure": self._structure.read(),
+            "grid": self.grid(),
+            "bands": self.bands(),
+            "kpoints": self.kpoints(),
+            "partial_charge": parchg,
+        }
+
+    @_base.data_access
+    def to_stm(
+        self,
+        selection: str = "constant_height",
+        tip_height: float = 2.0,
+        current: float = 1.0,
+        supercell: Union[int, np.array] = 2,
+    ) -> Graph:
+        """Generate STM image data from the partial charge density.
+
+        Parameters
+        ----------
+        selection : str
+            The mode in which the STM is operated and the spin channel to be used.
+            Possible modes are "constant_height"(default) and "constant_current".
+            Possible spin selections are "total"(default), "up", and "down".
+        tip_height : float
+            The height of the STM tip above the surface in Angstrom.
+            The default is 2.0 Angstrom. Only used in "constant_height" mode.
+        current : float
+            The tunneling current in nA. The default is 1.
+            Only used in "constant_current" mode.
+        supercell : int | np.array
+            The supercell to be used for plotting the STM. The default is 2.
+
+        Returns
+        -------
+        Graph
+            The STM image as a graph object. The title is the label of the Contour
+            object.
+        """
+        _raise_error_if_vacuum_too_small(self._estimate_vacuum())
+
+        tree = select.Tree.from_selection(selection)
+        for index, selection in enumerate(tree.selections()):
+            if index > 0:
+                message = "Selecting more than one STM is not implemented."
+                raise exception.NotImplemented(message)
```

Comment:
The Graph object can handle multiple Contour plots. You will just only see one of them so perhaps it is still good not to allow more than one.

Reply:
I agree... better to leave the NotImplemented exception in the code I think. This was actually your addition :)